inc/common/purge.php

```diff
@@ -124,24 +113,31 @@ function rocket_get_purge_urls( $post_id, $post ) {
 		}
 
 		foreach ( $cache_purge_pages as $page ) {
-			$page = trailingslashit( $home_url ) . $page;
-			array_push( $purge_urls, $page );
+			// Check if it contains regex pattern.
+			if ( strstr( $page, '*' ) ) {
+				$matches_files = _rocket_get_recursive_dir_files_by_regex( '#' . $page . '#i' );
+				foreach ( $matches_files as $file ) {
+					// Convert path to URL.
+					$purge_urls[] = str_replace( rocket_get_constant( 'WP_ROCKET_CACHE_PATH' ), untrailingslashit( $home_url ) . '/', $file->getPath() );
```

Comment:
This string replacement doesn't give the expected result:

Using `/category/(.*)`, here is the result:
```
Before: /app/public/wp-content/cache/wp-rocket/example.org/category/unpublished
After:    https://example.org/example.org/category/unpublished
```

Why?

Because `WP_ROCKET_CACHE_PATH` returns the filepath for the `cache/wp-rocket/` folder, but not the one for the current domain.

One solution is to do this before the first foreach:
`$cache_path = rocket_get_constant( 'WP_ROCKET_CACHE_PATH' ) . wp_parse_url( $home_url, PHP_URL_HOST );`

And then

`str_replace( $cache_path, untrailingslashit( $home_url ), $file->getPath() );`

Reply:
Done